### ISSUE TITLE
twisted[tls] extra probably ought to include idna package as well

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7853]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#7853|
|Type|enhancement|
|Created|2015-04-15 04:05:36Z|
|Branch|https://github.com/twisted/twisted/tree/tls-idna-req-7853|

The idna package is required for properly interpreting idna2008 names, so certain certificates won't verify properly without it.  Probably not a lot, but it's still worth putting into the extra.

Attachments:

* [add-idna-tls-extra-7853-1.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/91e/91e11ad5c05ed928a3f6978680599a5a25b8dba4/c26b53a887a9c1be024d68b09a90f46202d24040.patch) (2003 bytes) - added by herrwolfe on 2015-04-15 12:37:45Z - add idna to extras, tests
* [add-idna-tls-extra-7853-2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/91e/91e11ad5c05ed928a3f6978680599a5a25b8dba4/89de5bee762a004ddcaac3267b4911a31133bda1.patch) (2258 bytes) - added by herrwolfe on 2015-04-15 12:46:33Z - add topfile
* [add-idna-tls-extra-7853-3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/91e/91e11ad5c05ed928a3f6978680599a5a25b8dba4/41f5a8e3fcad87d43c2ae4c4614026e7c6d88d88.patch) (3104 bytes) - added by herrwolfe on 2015-04-15 17:49:27Z - add docs, set git config to use noprefix
* [add-idna-tls-extra-7853-4.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/91e/91e11ad5c05ed928a3f6978680599a5a25b8dba4/e3a6b3d90fbb92f37ac4bb3212938477ac3befe8.patch) (3156 bytes) - added by herrwolfe on 2015-04-15 17:52:03Z - fix newsfile

<details><summary>Searchable metadata</summary>

```
trac-id__7853 7853
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__None None
branch__branches_tls_idna_req_7853 branches/tls-idna-req-7853
branch_author__adiroiban adiroiban
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1429070736649846 1429070736649846
changetime__1445217226370668 1445217226370668
version__None None
owner__herrwolfe herrwolfe

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_1"></a>|@derwolfe commented|
|-|-|

The patch https://twistedmatrix.com/trac/attachment/ticket/7853/add-idna-tls-extra-7853-2.patch adds ``` idna ``` to the ``` tls ``` extra.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

(In [44454]) Branching to tls-idna-req-7853.

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_3"></a>|@derwolfe commented|
|-|-|

Also - for a bit of a smoke test, you can run ``` pip install ".[tls]" ``` from the twisted root directory (where setup.py lives). You should see ``` idna ``` downloaded during the install process.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set owner to @derwolfe|
|-|-|

Thanks for the patch.

We document what is installed in ```docs/installation/howto/optional.rst``` as it is not available in user-facing documentation otherwise (eg. on PyPI).

Please add the idna package to the documentation and resubmit for review.

-hawkie

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

Beside the documentation, I think that the news fragment need to be improved.
It read more like a commit message and not like release notes.

Maybe something like this:

The tls optional dependency will now also install the idna package to validate idna2008 names.

Other than that it looks good.

I have tested on my branch and idna is pulled.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_6"></a>|@derwolfe commented|
|-|-|

Adiroiban, hawkie

Thanks for your comments. I'll update the patch this evening. I'll also reconfigure git so my patches work (by using ``` git config --file Twisted/.git/config --bool --add diff.noprefix true ```)

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="derwolfe's avatar" src="https://avatars.githubusercontent.com/u/641357?s=50" width="50" height="50">](https://github.com/derwolfe)<a name="note_7"></a>|@derwolfe removed owner|
|-|-|

The new patch, https://twistedmatrix.com/trac/attachment/ticket/7853/add-idna-tls-extra-7853-4.patch, includes:

1. changes to the installation docs; they now state that the tls extra installs idna support, link to pypi.
2. the changes to the t.p.d._EXTRA_OPTIONS and the tests
3. the updated newsfile (per adiroiban's suggestions).

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban set owner to @derwolfe|
|-|-|

Looks good.

I have sent it to buildbot... if pass will merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_9"></a>|@adiroiban set status to `closed`|
|-|-|

(In [44523]) Merge tls-idna-req-7853: Add idna as tls dependency.

Author: herrwolfe
Reviewers: hawkowl, adiroiban
Fixes: [#7853](https://github.com/twisted/twisted/issues/7853)

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_10"></a>|@hawkowl commented|
|-|-|

Ticket retargeted after milestone deleted
